tests/end2end/test_collection_view.py

```diff
@@ -39,3 +43,100 @@ def test_user_sees_error_message_if_collection_code_does_not_exist(user, testapp
     res = testapp.get(url_for('collection.view', collection_code='FAKE1')).follow()
     # Sees error message.
     assert escape(_('Collection code "%(code)s" does not exist', code='FAKE1')) in res
+
+
+def test_superuser_can_view_all_permissions_on_collection(superuser, permission, testapp):
+    """View all permissions as superuser."""
+    # Goes to homepage.
+    res = testapp.get('/')
+    # Fills out login form.
+    form = res.forms['loginForm']
+    form['username'] = superuser.email
+    form['password'] = 'myPrecious'
+    # Submits.
+    res = form.submit().follow()
+    assert res.status_code is 200
+    # Goes to the right URL for viewing a collection.
+    res = testapp.get(url_for('collection.view', collection_code=permission.collection.code))
+    assert res.status_code is 200
+    # Sees all permissions.
+    assert _('Permissions') in res
+    assert permission.user.email in res
+
+
+def test_cataloging_admin_can_view_all_permissions_on_own_collection(user, collection, testapp):
+    """View all permissions on a collection that you're managing."""
+    # Add cataloging admin permission.
+    cataloging_admin_permission = PermissionFactory(user=user, collection=collection,
+                                                    cataloging_admin=True)
+    cataloging_admin_permission.save()
+    # Add 2nd cataloger+registrant permission.
+    other_users_non_cataloging_admin_permission = PermissionFactory(collection=collection,
```

Comment:
Maybe we should add an extra permission with another collection here, to make it more clear that it's the collection that decides if we can see the permission or not? Or maybe that's covered sufficiently by other tests?

Reply:
We're doing that in the new tests added in latest push I think, e.g. `test_cataloging_admin_sees_only_cataloging_admins_on_others_collection`. I think that it's pretty clear with the 6-7 tests on the same topic that I added today!